internal/manifests/queryfrontend/oauth_proxy.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		return nil, err
+	}
+
+	labels := manifestutils.ComponentLabels(manifestutils.OAuthProxyPortName, tempo.Name)
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cookieSecretName(tempo),
+			Labels:    labels,
+			Namespace: tempo.Namespace,
+		},
+		Data: map[string][]byte{
+			sessionSecretKey: []byte(sessionSecret),
+		},
+	}, nil
+}
+
+func cookieSecretName(tempo v1alpha1.TempoStack) string {
+	return fmt.Sprintf("tempo-%s-cookie-proxy", tempo.Name)
+}
+
+func proxyInitArguments(tempo v1alpha1.TempoStack) []string {
+
+	return []string{
+		fmt.Sprintf("--cookie-secret-file=%s/%s", oauthProxySecretMountPath, sessionSecretKey),
+		fmt.Sprintf("--https-address=:%d", manifestutils.OAuthProxyPort),
+		fmt.Sprintf("--openshift-service-account=%s", naming.Name(manifestutils.QueryFrontendComponentName, tempo.Name)),
+		"--provider=openshift",
+		fmt.Sprintf("--tls-cert=%s/tls.crt", tlsProxyPath),
+		fmt.Sprintf("--tls-key=%s/tls.key", tlsProxyPath),
+		fmt.Sprintf("--upstream=http://localhost:%d", manifestutils.PortJaegerUI),
+	}
+}
+
+func patchDeploymentForOauthProxy(params manifestutils.Params, dep *v1.Deployment) {
+	tempo := params.Tempo
+	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: getTLSSecretNameForFrontendService(tempo),
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: getTLSSecretNameForFrontendService(tempo),
+			},
+		},
+	})
+	dep.Spec.Template.Spec.ServiceAccountName = naming.Name(manifestutils.QueryFrontendComponentName, tempo.Name)
+	dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, oAuthProxyContainer(params))
+	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: cookieSecretName(tempo),
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: cookieSecretName(tempo),
+			},
+		},
+	})
+}
+
+func oAuthProxyContainer(params manifestutils.Params) corev1.Container {
+	tempo := params.Tempo
+	args := proxyInitArguments(tempo)
+
+	if len(strings.TrimSpace(tempo.Spec.Template.QueryFrontend.JaegerQuery.Oauth.SAR)) > 0 {
+		args = append(args, fmt.Sprintf("--openshift-sar=%s", tempo.Spec.Template.QueryFrontend.JaegerQuery.Oauth.SAR))
+	}
+
+	oauthProxyImage := tempo.Spec.Images.OauthProxy
+	if oauthProxyImage == "" {
+		oauthProxyImage = params.CtrlConfig.DefaultImages.OauthProxy
+	}
+	return corev1.Container{
+		Image: oauthProxyImage,
+		Name:  "oauth-proxy",
+		Args:  args,
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: manifestutils.OAuthProxyPort,
+				Name:          manifestutils.OAuthProxyPortName,
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{{
+			MountPath: tlsProxyPath,
+			Name:      getTLSSecretNameForFrontendService(tempo),
+		},
+
+			{
+				MountPath: oauthProxySecretMountPath,
+				Name:      cookieSecretName(tempo),
+			},
+		},
+		Resources: resources(tempo),
```

Comment:
That will use the resources defined for `tempo.Spec.Template.QueryFrontend.Resources`?

Reply:
For now yes :/ is the same case for jaegerUI/tempo-querier